packages/wallet-manager/src/wallets/Evm/Evm.ts

```diff
@@ -0,0 +1,164 @@
+import { ethers } from 'ethers';
+import events from 'events';
+import type {
+  Web3Provider,
+  ExternalProvider,
+  Provider
+} from '@ethersproject/providers';
+import { customEVMEvents, IEvmWallet } from '../../interfaces';
+import { AddChain } from '../../types';
+import { checkWindow } from '../../utils';
+
+class EvmWallet extends events.EventEmitter implements IEvmWallet {
+  address?: string;
+  signer?: ethers.Signer;
+  web3Provider!: Web3Provider;
+
+  constructor(provider?: Web3Provider) {
+    super();
+
+    if (!window.ethereum) {
+      throw new Error('window.ethereum is not defined.');
+    }
+    if (provider) {
+      this.web3Provider = provider;
+    } else {
+      this.web3Provider = new ethers.providers.Web3Provider(
+        window.ethereum as ExternalProvider
+      );
+    }
+    this.appendProviderEvents();
+  }
+
+  /**
+   * @name calculateAccountData
+   * @param accounts
+   * @returns {Promise<void>}
+   */
+  private async resetAccounts(accounts?: string[]): Promise<void> {
+    if (accounts?.length) {
+      this.address = accounts[0];
+    }
+
+    const _accounts = await this.web3Provider.listAccounts();
+
+    if (!_accounts.length) {
+      return;
+    }
+
+    this.address = _accounts![0];
+    this.signer = this.web3Provider.getSigner();
+  }
+
+  /**
+   * @name reconnectToProvider
+   * @returns {void}
+   */
+  private reconnectToProvider(): void {
+    this.web3Provider = new ethers.providers.Web3Provider(
+      window.ethereum as ExternalProvider
+    );
+  }
+
+  /**
+   * @name appendProviderEvents
+   * @returns {void}
+   * @description Appends the provider events to the windowConnector
+   */
+  private appendProviderEvents(): void {
+    checkWindow();
+
+    (this.web3Provider.provider as Provider).on('connect', async () => {
+      this.reconnectToProvider();
+      await this.resetAccounts();
+    });
+
+    (this.web3Provider.provider as Provider).on('disconnect', async () => {
+      this.reconnectToProvider();
+      await this.resetAccounts();
+    });
+
+    (this.web3Provider.provider as Provider).on('chainChanged', async () => {
+      this.reconnectToProvider();
+      await this.resetAccounts();
+
+      this.emit(customEVMEvents.CHAIN_CHANGE, this.web3Provider);
+    });
+
+    (this.web3Provider.provider as Provider).on(
+      'accountsChanged',
+      async (accounts: string[]) => {
+        this.reconnectToProvider();
+        await this.resetAccounts(accounts);
+
+        this.emit(customEVMEvents.ACCOUNT_CHANGE, this.address);
+      }
+    );
+  }
+  /**
+   * @name connect
+   * @description Connects the wallet to the provider
+   * @throws {Error} if the window object is not defined
+   * @returns {Promise<void>}
+   */
+  public async connect() {
+    try {
+      checkWindow();
+    } catch (e) {
+      throw e;
+    }
```

Comment:
you can just write `checkWindow()` and the behaviour will be the same

Reply:
my mistake forgot to fix that ocurrence